### ISSUE TITLE
Fix false positives for valueless attribute selectors in selector-attribute-name-disallowed-list

### ISSUE THREAD

**Issue by vkrol** [stylelint/stylelint#5059]
> Clearly describe the bug

`selector-attribute-name-disallowed-list` doesn't handle valueless attribute selectors correctly.

> Which rule, if any, is the bug related to?

`selector-attribute-name-disallowed-list`.

> What code is needed to reproduce the bug?

```css
[disabled] { }
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-attribute-name-disallowed-list": ["disabled"]
  }
}
```

> Which version of stylelint are you using?

`13.8.0`.

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.css" --config myconfig.json`.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

The warning was flagged.

> What actually happened (e.g. what warnings or errors did you get)?

No warnings were flagged.

/cc @rletsin
